cassandra/src/main/java/io/micronaut/configuration/cassandra/health/CassandraHealthIndicator.java

```diff
@@ -0,0 +1,154 @@
+import java.net.InetSocketAddress;
+import java.util.*;
+
+/**
+ * A {@link io.micronaut.management.health.indicator.HealthIndicator} for Cassandra.
+ *
+ * @author Ilkin Ashrafli
+ * @since 2.1.0
+ */
+@Requires(property = HealthEndpoint.PREFIX + ".cassandra.enabled", notEquals = "false")
+@Requires(beans = HealthEndpoint.class)
+@Singleton
+public class CassandraHealthIndicator extends AbstractHealthIndicator<Map<String, Object>> {
+
+    private static final String COL_RELEASE_VERSION = "release_version";
+    private static final String COL_CLUSTER_NAME = "cluster_name";
+    private static final String COL_CQL_VERSION = "cql_version";
+    private static final String COL_DATA_CENTER = "data_center";
+    private static final String COL_NATIVE_PROTOCOL_VERSION = "native_protocol_version";
+    private static final String COL_PARTITIONER = "partitioner";
+    private static final String COL_RACK = "rack";
+
+    private static final SimpleStatement VALIDATION_SELECT = selectFrom("system", "local")
+            .column(COL_RELEASE_VERSION)
+            .column(COL_CLUSTER_NAME)
+            .column(COL_CQL_VERSION)
+            .column(COL_DATA_CENTER)
+            .column(COL_NATIVE_PROTOCOL_VERSION)
+            .column(COL_PARTITIONER)
+            .column(COL_RACK)
+            .build()
+            .setConsistencyLevel(ConsistencyLevel.LOCAL_ONE);
+
+    private static final Logger LOG = LoggerFactory.getLogger(CassandraHealthIndicator.class);
+
+    private final CqlSession cqlSession;
+
+    /**
+     * Default constructor.
+     *
+     * @param cqlSession The The cassandra {@link CqlSession} to query for details
+     */
+    public CassandraHealthIndicator(final CqlSession cqlSession) {
+        this.cqlSession = cqlSession;
+    }
+
+    @Override
+    protected Map<String, Object> getHealthInformation() {
+        Map<String, Object> detail = new LinkedHashMap<>();
+        Map<UUID, Node> nodes = cqlSession.getMetadata().getNodes();
+        detail.put("session", cqlSession.isClosed() ? "CLOSED" : "OPEN");
+        Optional<CqlIdentifier> opKeyspace = cqlSession.getKeyspace();
+        if (opKeyspace.isPresent()) {
+            detail.put("keyspace", opKeyspace.get());
+        }
+        detail.put("nodes_count", nodes.keySet().size());
+
+        Map<UUID, Map<String, Object>> nodesMap = new HashMap<>();
+        Map<NodeState, Integer> nodeStateMap = new EnumMap<>(NodeState.class);
+        boolean up = false;
+        int i = 0;
+        for (Map.Entry<UUID, Node> entry : nodes.entrySet()) {
+            UUID uuid = entry.getKey();
+            Node node = entry.getValue();
+            nodeStateMap.merge(node.getState(), 1, (a, b) -> a + b);
+            if (node.getState() == NodeState.UP) {
+                up = true;
+            }
+            if (i++ < 10) {
+                Map<String, Object> nodeMap = new HashMap<>();
+                Optional<InetSocketAddress> opBroadcastAddress = node.getBroadcastAddress();
```

Comment:
You should also mention the endpoint, it's even more important than the broadcast address:

```
nodeMap.put("endpoint", node.getEndPoint());
```

Endpoint and broadcast address are usually the same, but not always.